lib/syskit/test/network_manipulation.rb

```diff
@@ -160,11 +161,21 @@ def syskit_generate_network(*to_instanciate, add_missions: true)
                 end
                 task_mapping = plan.in_transaction do |trsc|
                     engine = NetworkGeneration::Engine.new(plan, work_plan: trsc)
-                    mapping = engine.compute_system_network(
+                    mapping, resolution_errors = engine.compute_system_network(
                         tasks_to_instanciate.map(&:planning_task),
                         validate_generated_network: false,
                         early_deploy: false
                     )
+                    unless resolution_errors.empty?
+                        resolution_errors.each do |error|
+                            t = error.planned_task
+                            expect_execution do
+                                t.failed_event.emit(error.original_exceptions)
+                            end
+                        end
```

Comment:
Would be better to invert, that is run the loop inside the expect_execution

And replace expect_execution by execute. expect_execution without a `to` block actually does nothing